.gitignore

```diff
@@ -21,6 +22,10 @@ data/**
 log/
 *.egg-info
 
+# uv
+# We may want to not ignore this in the future to lock versions across fresh builds
+uv.lock
```

Comment:
I think this shouldn't be gitignored:

https://docs.astral.sh/uv/guides/projects/#uvlock

> This file should be checked into version control, allowing for consistent and reproducible installations across machines.

Please be more careful when editing these files!

Reply:
Usually lock files are not ignored, but given that a `uv.lock` does not exist in tree, I thought that we weren't using `uv`, and thus thought that I should ignore `uv.lock` to not impose version requirements right away.

Will remove.

Reply:
Removed.